crates/cli/src/config.rs

```diff
@@ -790,86 +852,96 @@ impl Config {
-
+    fn load_proj_config() -> RawConfig {
         // TODO(cloutiertyler): For now we're checking for a spacetime.toml file
         // in the current directory. Eventually this should really be that we
         // search parent directories above the current directory to find
         // spacetime.toml files like a .gitignore file
         let cur_dir = std::env::current_dir().expect("No current working directory!");
-        let cur_config = Self::load_raw(cur_dir, true);
+        if let Some(config_path) = Self::find_config_path(&cur_dir) {
+            Self::load_from_file(&config_path)
+                .inspect_err(|e| eprintln!("config file {config_path:?} is invalid: {e:#}"))
+                .unwrap_or_default()
+        } else {
+            Default::default()
+        }
+    }
+
+    pub fn load() -> Self {
+        let home_path = Self::system_config_path();
+        let home_lock = Lockfile::for_config(&home_path).unwrap();
+        let home = if home_path.exists() {
+            Self::load_from_file(&home_path)
+                .inspect_err(|e| eprintln!("config file {home_path:?} is invalid: {e:#?}"))
+                .unwrap_or_default()
+        } else {
+            RawConfig::new_with_localhost()
+        };
+
+        let proj = Self::load_proj_config();
 
         Self {
-            home: home_config,
-            proj: cur_config,
+            home,
+            proj,
+
+            home_file: Some((home_path, home_lock)),
         }
     }
 
+    #[doc(hidden)]
+    /// Used in tests; not backed by a file.
     pub fn new_with_localhost() -> Self {
         Self {
             home: RawConfig::new_with_localhost(),
             proj: RawConfig::default(),
+
+            home_file: None,
         }
     }
 
-    pub fn save(&self) {
-        let config_var = std::env::var_os("SPACETIME_CONFIG_FILE");
-        let config_var_exists = config_var.is_some();
-        let config_path = if let Some(config_path) = config_var {
-            PathBuf::from(config_path)
-        } else {
-            let home_dir = dirs::home_dir().unwrap();
-            let config_dir = home_dir.join(HOME_CONFIG_DIR);
-            if !config_dir.exists() {
-                fs::create_dir_all(&config_dir).unwrap();
-            }
+    #[doc(hidden)]
+    pub fn clone_for_test(&self) -> Self {
+        assert!(
+            self.home_file.is_none(),
+            "Cannot clone_for_test a Config derived from file {:?}",
+            self.home_file,
+        );
+        Config {
+            home: self.home.clone(),
+            proj: self.proj.clone(),
+            home_file: None,
+        }
+    }
 
-            Self::find_config_path(&config_dir).unwrap_or_else(|| config_dir.join(CONFIG_FILENAME))
+    pub fn save(&self) {
+        let Some((home_path, _)) = &self.home_file else {
+            return;
         };
 
+        let parent = home_path.parent().unwrap();
+        if parent != Path::new("") {
```

Comment:
why did this become a `parent != Path::new("")` instead of a `parent.exists()`?

Reply:
- `make_dir_all` on an existing directory is a no-op, so we don't need to check for that.
- `make_dir_all` on the empty path is invalid, so we do need to avoid doing that to handle the case where the config path is a relative path with no directory component.

Reply:
Encapsulated in a helper function with comments.